src/wallet/wallet.cpp

```diff
@@ -1440,8 +1434,12 @@ void CWallet::BlockDisconnected(const CBlock& block) {
     auto locked_chain = chain().lock();
     LOCK(cs_wallet);
 
+    // At block disconnection, this will change an abandoned transaction to
+    // be unconfirmed, whether or not the transaction is added back to the mempool.
+    // User may have to call abandontransaction again. It may be addressed in the
+    // future with a stickier abandoned state or even removing abandontransaction call.
```

Comment:
In commit "Encapsulate tx status in a Confirmation struct" (a31be09bfd77eed497a8e251d31358e16e2f2eb1)

Comment seems slightly off. I think the comment is right that `abandontransaction` would need to be called again in this case, but I don't think it would just be because of this code. The abandoned state would already have been lost previously when the block was connected.